### DISCUSSION TITLE
InitialData Cache - Next.js SSR

### DISCUSSION THREAD

**Discussion by tomevans18** [vercel/swr#447] [General]
Following reviewing a PR around `initialData` cache (https://github.com/vercel/swr/pull/388) I now understand that `initialData` is actually a fallback and will be set to the key within the cache if its undefined. This means if there is an existing cache against said key, then `initialData` is effectively ignored.

The sadly this creates an issue of stale data. Within our situation we have:
- a page where a user can update a list
- a page where the user can view another list (different API and key) but this list accesses some of this data from the other page.

When the second page loads, if there is an existing cache, the server will call the API and obtain the latest data, which is then passed to `initialData` but as there is a cached version, `initialData` is ignored and now the data loaded is out of date.

Has anyone found a way around this?

We have tried adding `mutate([secondPageSWRKey])` but that doesn't seem to work. We assume this is because you need to pass the data in the second arg if the `swr` setup/method is on another page but this would mean making another API call to an endpoint thats is not truly related to the component, which seems a little wrong to do.

**Comment by shuding** [comment]
Currently yes, `initialData` works just like a fallback that almost equivalents to:

```js
const data = useSWR(...).data || initialData
```

And we have an improved solution to make it a real _inital value_, not just a fallback. The new API is planned and we are gonna work on it soon.

I shared it on https://github.com/vercel/swr/pull/388#issuecomment-638085648, in the **API Design** part. Ideally it should be something like:

```js
<SWRConfig value={{ initialData: { [key]: data } }}>
  <Component/>
</SWRConfig>
```

**Reply by tomevans18** [reply]
Yes, I saw that comment but perhaps misunderstood it.
We use SWRConfig at the `_app` level and my understanding was this is a global config but are you saying this can also be used at the component level too, effectively merging with the global config?
